ballerina/client.bal

```diff
@@ -341,6 +341,192 @@ public isolated client class Client {
+            @display {label: "Completed Parts"} CompletedPart[] completedParts)
+            returns error? {
+        
+        if objectName == EMPTY_STRING {
+            return error(EMPTY_OBJECT_NAME_ERROR_MSG);
+        }
+        if bucketName == EMPTY_STRING {
+            return error(EMPTY_BUCKET_NAME_ERROR_MSG);
+        }
+
+        http:Request request = new;
+
+        string requestURI = string `/${bucketName}/${objectName}`;
+        string queryParamStr = string `?uploadId=${uploadId}`;
+        map<string> queryParamsMap = {
+            "uploadId": uploadId
+        };
+        map<string> requestHeaders = setDefaultHeaders(self.amazonHost);
+
+        check generateSignature(self.accessKeyId, self.secretAccessKey, self.region, POST, requestURI, 
+            UNSIGNED_PAYLOAD, requestHeaders, request, queryParams = queryParamsMap);
+        requestURI = string `${requestURI}${queryParamStr}`;
+
+        string payload = string `<CompleteMultipartUpload xmlns="http://s3.amazonaws.com/doc/2006-03-01/">`;
+        foreach CompletedPart part in completedParts {
+            payload = string `${payload}<Part><PartNumber>${part.partNumber.toString()}</PartNumber><ETag>${part.ETag}</ETag></Part>`;
+        }
+        payload = string `${payload}</CompleteMultipartUpload>`;
+        request.setPayload(payload);
+
+        http:Response httpResponse = check self.amazonS3->post(requestURI, request);
+        return handleHttpResponse(httpResponse);
+    }
+
+    # Aborts a multipart upload.
+    #
+    # + objectName - The name of the object
+    # + bucketName - The name of the bucket 
+    # + uploadId - The upload ID of the multipart upload
+    # + return - An error on failure or else `()`
+    remote isolated function abortMultipartUpload(
+            @display {label: "Object Name"} string objectName,
+            @display {label: "Bucket Name"} string bucketName,
+            @display {label: "Upload ID"} string uploadId)
+            returns error? {
+        
+        if objectName == EMPTY_STRING {
+            return error(EMPTY_OBJECT_NAME_ERROR_MSG);
+        }
+        if bucketName == EMPTY_STRING {
+            return error(EMPTY_BUCKET_NAME_ERROR_MSG);
+        }
+
+        map<string> queryParamsMap = {
+            "uploadId": uploadId
+        };
+
+        string queryParamStr = string `?uploadId=${uploadId}`;
+        http:Request request = new;
+
+        string requestURI = string `/${bucketName}/${objectName}`;
+        map<string> requestHeaders = setDefaultHeaders(self.amazonHost);
+
+        check generateSignature(self.accessKeyId, self.secretAccessKey, self.region, DELETE, requestURI, 
+            UNSIGNED_PAYLOAD, requestHeaders, request, queryParams = queryParamsMap);
+        requestURI = string `${requestURI}${queryParamStr}`;
+
+        http:Response httpResponse = check self.amazonS3->delete(requestURI, request);
+        return handleHttpResponse(httpResponse);
```

Comment:
```suggestion
        http:Request request = new;
        string requestURI = string `/${bucketName}/${objectName}?uploadId=${uploadId}`;
        map<string> requestHeaders = setDefaultHeaders(self.amazonHost);

        check generateSignature(self.accessKeyId, self.secretAccessKey, self.region, DELETE, requestURI, 
            UNSIGNED_PAYLOAD, requestHeaders, request, queryParams = queryParamsMap);

        http:Response httpResponse = check self.amazonS3->delete(requestURI, request);
        return handleHttpResponse(httpResponse);
```

Reply:
In this case `requestURI` shouldn't contain query parameters generating the signature, only the resource being accessed. That is why I added the query parameter after the signature was made.

Reply:
Is the `requestURI` is being used when generating the signature ? If not what is the extra advantage of defining two variables which will be concatenated to one at the end ?

Reply:
I agree there's no advantage in defining two variables. 
I have modified it to concatenate directly without being defined in another variable